projects/demo-cypress/src/tests/range.cy.ts

```diff
@@ -0,0 +1,275 @@
+            });
+        });
+
+        describe('keySteps', () => {
+            beforeEach(() => {
+                component.keySteps = [
+                    [0, 0],
+                    [25, 10_000],
+                    [50, 100_000],
+                    [75, 500_000],
+                    [100, 1_000_000],
+                ];
+                component.step = (component.max - component.min) / 10;
+                component.testValue.setValue([0, 0]);
+                fixture.detectChanges();
+            });
+
+            const testsContexts = [
+                {
+                    value: [0, 10_000],
+                    leftOffset: '0%',
+                    rightOffset: '75%',
+                },
+                {
+                    value: [10_000, 10_000],
+                    leftOffset: '25%',
+                    rightOffset: '75%',
+                },
+                {
+                    value: [10_000, 100_000],
+                    leftOffset: '25%',
+                    rightOffset: '50%',
+                },
+                {
+                    value: [100_000, 100_000],
+                    leftOffset: '50%',
+                    rightOffset: '50%',
+                },
+                {
+                    value: [100_000, 500_000],
+                    leftOffset: '50%',
+                    rightOffset: '25%',
+                },
+                {
+                    value: [500_000, 500_000],
+                    leftOffset: '75%',
+                    rightOffset: '25%',
+                },
+                {
+                    value: [500_000, 750_000],
+                    leftOffset: '75%',
+                    rightOffset: '12.5%',
+                },
+                {
+                    value: [750_000, 1_000_000],
+                    leftOffset: '87.5%',
+                    rightOffset: '0%',
+                },
+            ] as const;
+
+            testsContexts.forEach(({value, leftOffset, rightOffset}) => {
+                it(`${JSON.stringify(value)}`, () => {
+                    component.testValue.setValue(value as unknown as number[]);
+                    fixture.detectChanges();
+
+                    expect(getFilledRangeOffset(component).left).to.equal(leftOffset);
+                    expect(getFilledRangeOffset(component).right).to.equal(rightOffset);
+                });
+            });
+        });
+    });
+
+    interface Offset {
+        left: string;
+        right: string;
+    }
+
+    function getFilledRangeOffset(component: TestComponent): Offset {
+        const computedStyles = component.el.nativeElement;
+
+        return {
+            left: getComputedStyle(computedStyles).getPropertyValue('--left'),
+            right: getComputedStyle(computedStyles).getPropertyValue('--right'),
+        };
+    }
+
+    function getLeft(component: TestComponent): HTMLElement | null {
+        return (
+            component.el.nativeElement.querySelector(
```

Comment:
It not unit test. Let's use framework API instead of direct access to DOM

Reply:
fixed